### PR TITLE
Additional modules fixes #194

### LINKED ISSUE #194: Support for additional modules
Currently if someone decides to write their own modules they can do it and just put them in a directory for the given module. 
But running `go generate` will change `types.go` in order for the modules to be imported and useful. This means that it will be harder to do other changes and not accidentally commit `types.go`. 

I propose that we add the possibility for the module_generator to inspect a predefined gitignored file, `additional.list`, with list of package names to be included in `additional_types.go`.

Illustrating example: 
>
> inside module_dir/additional.list:
> ```go
> example.com/modules/awesome
> example.com/modules/super
> ```
> 
> inside module_dir/new.go
> ```go
> // module_dir documentation
> package module_dir
> 
> //go:generate go run ../tools/module_generator/main.go -template "types.go.template" -output  "types.go"
>
> //go:generate go run ../tools/module_generator/main.go -inputlist additional.list -template "additional_types.go.template" -output "additional_types.go"
> ```

* will give all the sub directories inside `module_dir` to `types.go.template` as arguments
* will give the contents of `additional.list` to `additional_types.go.template` as arguments

This means that the changes will be to the:
* module_generator
* all `new.go` files
* new templates `additional_types.go.template`
* .gitignore (adding `additional.list`, `additional_types.go`)
* maybe rewrite the `types.go.template` and move everything, but adding to the map, in `new.go`

### PR DESCRIPTION
**Pull request by MStoykov**

### CODE REVIEW

**Comment by MStoykov** [comment]
If this changes seem OK, the same will be done for the cache, storage, logger and the two types of upstream balancing modules.

File: handler/additional_types.go.template [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"github.com/ironsmile/nedomi/config"
+	"github.com/ironsmile/nedomi/types"
+
+    {{range .}}"{{.}}"
+{{end}}
+)
+
+var _ = func() (s struct{}){
```
Comment (ironsmile) [inline handler/additional_types.go.template]: Is there a particular reason why this must be a function call? Wouldn't it work just the same if it was in the global scope?

Comment (MStoykov) [inline handler/additional_types.go.template]: yes there is a reason :smiley:  it doesn't work in any other way. 
I could've done it in `init` but there can only be one `init` function so ... I choose this method. Little bit javascripty if you ask me :laughing:

Comment (ironsmile) [inline handler/additional_types.go.template]: Ok, that's fine then. It is computer generated code anyway, so no one can expect it to be pretty!

**Pull request merged**